src/jit/lowerxarch.cpp

```diff
@@ -2299,6 +2299,18 @@ void Lowering::TreeNodeInfoInitLogicalOp(GenTree* tree)
         // as reg optional.
         SetRegOptionalForBinOp(tree);
     }
+
+    if (varTypeIsByte(op1))
+    {
+        regMaskTP regMask;
+        bool      containedNode = (op2->gtLsraInfo.srcCount == 0) && (op2->gtLsraInfo.dstCount == 0);
+        if (!containedNode)
+        {
+            regMask = op2->gtLsraInfo.getSrcCandidates(l);
+            assert(regMask != RBM_NONE);
+            op2->gtLsraInfo.setSrcCandidates(l, regMask & ~RBM_NON_BYTE_REGS);
+        }
```

Comment:
I think that we need to be crisper about the invariants here. I, at least, am a bit confused about what cases we will encounter here. I think (I could be mistaken, and haven't investigated) that TYP_BYTE or TYP_UBYTE lclVars (and many subexpressions involving byte types) are effectively promoted to TYP_INT, with casts added by morph as needed to ensure proper semantics. Therefore, when we see these operations, they generally will be targeting something that is probably going to be stored to memory. @sivarv do I have that right? Otherwise, this would seem to be overly conservative because it is not checking whether we are in a situation that actually requires a byteable register.

---
In reply to: [81855970](https://github.com/dotnet/coreclr/pull/7466#discussion_r81855970) [](ancestors = 81855970)